components/composited/BudgetForm/BudgetForm.tsx

```diff
@@ -0,0 +1,89 @@
+import { useState, FormEventHandler, useRef, Fragment } from 'react';
+import { BudgetFields, BudgetSch } from '../../../validators/BudgetFields';
+import { Button } from '../../generic/Button/Button';
+import { ErrorMessage } from '../../generic/ErrorMessage/ErrorMessage';
+import { Input } from '../../generic/Input/Input';
+import { BudgetTypeRadio } from '../BudgetTypeRadio/BudgetTypeRadio';
+import { Select } from '../../generic/Select/Select';
+import { Textarea } from '../../generic/Textarea/Textarea';
+import { useUser } from '../../../hooks/useUser';
+
+type BudgetFormProps = {
+  onSubmit: (fields: BudgetFields) => void;
+};
+
+export const BudgetForm = ({ onSubmit }: BudgetFormProps) => {
+  const { data: user } = useUser();
+  const formRef = useRef<HTMLFormElement>(null);
+  const [inputErrorMessage, setInputErrorMessage] = useState(false);
+
+  const handleSubmit: FormEventHandler<HTMLFormElement> = (e) => {
+    e.preventDefault();
+
+    const formData = new FormData(e.currentTarget);
+
+    const result = BudgetSch.safeParse({
+      name: formData.get('name'),
+      description: formData.get('description'),
+      budgetType: formData.get('budgetType'),
+      estimatedCost: Number(formData.get('estimatedCost')),
+      estimatedRealizationDate: formData.get('estimatedRealizationDate'),
```

Comment:
We will probably create simple utility function that will handle convertion to currency format.